routing/src/fib/fibgroupstore.rs

```diff
@@ -452,10 +455,11 @@ pub mod tests {
         .unwrap();
 
         assert_eq!(fibroute.0.len(), 4);
-        assert_eq!(*fibroute.0[0].borrow(), *store.get(&key1).unwrap());
-        assert_eq!(*fibroute.0[1].borrow(), *store.get(&key2).unwrap());
-        assert_eq!(*fibroute.0[2].borrow(), *store.get(&key3).unwrap());
-        assert_eq!(*fibroute.0[3].borrow(), *store.get(&key4).unwrap());
+        assert_eq!(unsafe { &*fibroute.0[0].get() }, store.get(&key1).unwrap());
```

Comment:
Ugh, I definitely hate this.  Can we have some shuttle tests here to make sure we don't get any memory corruption?  This probably needs some sort of rethink, but there isn't time for that right now, so really good multi-threaded testing is in order here.